pkg/agent/manager/cache/cache.go

```diff
@@ -305,8 +305,6 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 		c.diffSelectors(existingEntry, newEntry, selAdd, selRem)
 		c.addSelectorIndicesRecord(selAdd, record)
 		c.delSelectorIndicesRecord(selRem, record)
-		notifySet.MergeSet(selAdd)
-		notifySet.MergeSet(selRem)
```

Comment:
Don't these need to be added to notifySets?

Reply:
Hmm, actually, seems like we need to add existingEntry.selectors and newEntry.selectors to notifySets? I need to think on this a little :laughing: gotta page all this back into my brain :)

Reply:
I had to do some paging as well. haha. This logic moved below to lines 330-342. The basic idea now is that if the selectors are different between the new and existing entry, then notify both selector sets. This part took a while for me to figure out but I think will notify everything that needs to be notified, without sending out unnecessary notifications.

Reply:
but there is a break on line 322, is it possible we get into a case where that case is called and notify set is not  updated?